tests/test_extract_coding.py

```diff
@@ -0,0 +1,228 @@
+import os
+import warnings
+
+from Bio.Seq import Seq
+from click.testing import CliRunner
+import pandas as pd
+import pandas.util.testing as pdt
+import pytest
+import screed
+
+
+@pytest.fixture
+def seq():
+    s = 'CGCTTGCTTAATACTGACATCAATAATATTAGGAAAATCGCAATATAACTGTAAATCCTGTTCTGTC'
+    with warnings.catch_warnings():
+        # Ignore The following warning because we don't use Bio.Alphabet
+        # explicitly:
+        # PendingDeprecationWarning: We intend to remove or replace
+        # Bio.Alphabet in 2020, ideally avoid using it explicitly in your
+        # code. Please get in touch if you will be adversely affected by this.
+        # https://github.com/biopython/biopython/issues/2046
+        warnings.simplefilter("ignore")
+        return Seq(s)
+
+
+@pytest.fixture
+def low_complexity_seq():
+    return "CCCCCCCCCACCACCACCCCCCCCACCCCCCCCCCCCCCCCCCCCCCCCCCACCCCCCCA" \
+           "CACACCCCCAACACCC"
+
+
+@pytest.fixture(params=['seq', 'low_complexity_seq'])
+def type_seq(request, seq, low_complexity_seq):
+    if request.param == 'seq':
+        return request.param, seq
+    elif request.param == 'low_complexity_seq':
+        return request.param, low_complexity_seq
+
+
+def test_three_frame_translation(seq):
+    from khtools.extract_coding import three_frame_translation
+
+    test = [str(x) for x in three_frame_translation(seq)]
+    true = [
+        'RLLNTDINNIRKIAI*L*ILFC', 'ACLILTSIILGKSQYNCKSCSV',
+        'LA*Y*HQ*Y*ENRNITVNPVL'
+    ]
+    assert test == true
+
+
+def test_compute_fastp_low_complexity(type_seq):
+    from khtools.extract_coding import compute_fastp_complexity
+
+    seqtype, seq = type_seq
+    test = compute_fastp_complexity(seq)
+    if seqtype == 'seq':
+        assert test == 0.746268656716418
```

Comment:
is it float32, comparing floats is tricky between operating systems etc. can you compare within a tolerance ?

Reply:
https://docs.python.org/dev/library/math.html#math.isclose